### PR TITLE
Remove non-libuv code from Node extension

### PR DESCRIPTION
**Pull request by murgatroid99**
Using the libuv code was already the default, and all of the 1.2.x artifacts use the libuv code. I have seen no bugs against 1.2.x that were specifically caused by the libuv implementation, so I think it is safe to remove the fallback to the other code.

This mainly just removes a few files that will be unused, but it also moves `completion_queue_uv.cc` to `completion_queue.cc` and merges the contents of `server_uv.cc` into `server.cc`.

### CODE REVIEW

File: tools/run_tests/run_tests.py [hunk 1]
```diff
@@ -490,8 +486,7 @@ def build_steps(self):
         # building for electron requires a patch version
         self.node_version += '.0'
       return [['tools/run_tests/helper_scripts/{}.sh'.format(build_script),
-               self.node_version,
```
Comment (apolcyn) [inline tools/run_tests/run_tests.py]: nit: I think this change also needs to be made on lines 479 and 480 above, in the windows case.

Comment (murgatroid99) [inline tools/run_tests/run_tests.py]: Done

File: src/node/ext/node_grpc.cc [hunk 2]
```diff
@@ -43,18 +43,15 @@
 #include "grpc/support/time.h"
 
 // TODO(murgatroid99): Remove this when the endpoint API becomes public
-#ifdef GRPC_UV
```
Comment (apolcyn) [inline src/node/ext/node_grpc.cc]: nit:

can the `ifdef GRPC_UV` on lines 438-440 in this file also be removed?

File: binding.gyp [hunk 3]
```diff
@@ -90,17 +87,11 @@
       'include'
     ],
     'defines': [
-      'GPR_BACKWARDS_COMPATIBILITY_MODE'
+      'GPR_BACKWARDS_COMPATIBILITY_MODE',
+      'GRPC_ARES=0',
+      'GRPC_UV'
```
Comment (apolcyn) [inline binding.gyp]: related to below comment: with all of the `ifdef GRPC_UV`'s removed, can this be taken out?

File: src/node/ext/node_grpc.cc (hunk 2 above)
Comment (murgatroid99) [inline src/node/ext/node_grpc.cc]: Done

File: binding.gyp (hunk 3 above)
Comment (murgatroid99) [inline binding.gyp]: No, `GRPC_UV` changes which TCP implementation is used. The `#ifdef`s in the extension code just piggyback on that.

**Comment by apolcyn** [comment]
Looks like one more `ifdef GRPC_UV` in the renamed `completion_queue.cc` file.

otherwise LGTM

**Comment by murgatroid99** [comment]
test this please

**Comment by murgatroid99** [comment]
The failures of Asan C and C++, the Bazel Basic build, and the performance tests were all caused by #10850, and actually passed. The Mac failure is #10582, #10368, #9727, and similar. The Windows failure is #9088. The Linux and Windows portability failures are being fixed in #10848. And the interop failure appears to be an infrastructure failure.

Jenkins: test this please. I'd like to see actual results for the interop tests.

**Comment by grpc-testing** [comment]
```
Performance differences noted:
Benchmark                                                                         cpu_time    real_time
------------------------------------------------------------------------------  ----------  -----------
BM_StreamCreateSendInitialMetadataDestroy<RepresentativeClientInitialMetadata>     +237.00      +237.00
```

**Comment by grpc-testing** [comment]
```
Performance differences noted:
Benchmark                                        cpu_time    real_time
---------------------------------------------  ----------  -----------
BM_CallCreateDestroy<InsecureChannel>              +73.50       +73.50
BM_CallCreateDestroy<LameChannel>                  +53.50       +53.50
BM_IsolatedCall_NoOp                               +50.50       +51.00
BM_IsolatedCall_Unary                              +91.50       +91.50
BM_IsolatedFilter<ServerDeadlineFilter, NoOp>       +6.00        +6.00
BM_LameChannelCallCreateCpp                      +1123.00     +1123.00
BM_MetadataFromInternedSlices                       +9.00        +9.00
BM_MetadataFromInternedSlicesAlreadyInIndex         +7.50        +7.50
BM_MetadataFromInternedSlicesWithBackingStore       +9.50        +9.50
BM_SliceReIntern                                   +14.50       +14.50
```

**Comment by grpc-testing** [comment]
```
Performance differences noted:
Benchmark         cpu_time    real_time
--------------  ----------  -----------
BM_SliceIntern      -23.00       -23.00
```

**Comment by murgatroid99** [comment]
Now the interop tests are passing.

**Pull request merged**